### PR TITLE
fix(1412): Aggregate event metrics daily

### PR DESCRIPTION
**Pull request by d2lam**
Related: https://github.com/screwdriver-cd/screwdriver/issues/1412

### CODE REVIEW

File: lib/helper.js [hunk 1]
```diff
@@ -48,7 +50,53 @@ function parseTemplateConfigName(config) {
     return {};
 }
 
+/**
+ * recursively fetching records (ex: builds/events) until the end
+ * @method getAllRecords
+ * @param  {String}       funcName    Function to fetch. For example: 'getBuilds'
+ * @param  {opts}         opts        opts to pass into scan function
```
Comment (tkyi) [inline lib/helper.js]:
```suggestion
 * @param  {Object}         opts        opts to pass into scan function
```
?

File: lib/helper.js [hunk 2]
```diff
@@ -48,7 +50,53 @@ function parseTemplateConfigName(config) {
...
+ * @param  {Array}        resultArray Result so far
+ * @param  {String}       date        Current date for aggregation
+ * @param  {Number}       index       Current index (each index is a different date)
+ * @return {Array}                    Array where each element is an array of records belong to the same date
+ */
+async function getAllRecords(funcName, opts, resultArray, date, index) {
+    const builds = await this[funcName](opts);
+    const length = builds.length;
```
Comment (tkyi) [inline lib/helper.js:65]: do you need make sure `builds` is an array?

File: lib/helper.js [hunk 3]
```diff
@@ -48,7 +50,53 @@ function parseTemplateConfigName(config) {
...
+        }
+
+        resultArray[currentIndex].push(currentBuild);
+    }
+    // last page
+    if (length < opts.paginate.count) {
+        return resultArray;
+    }
+
+    // might have more data, continue fetchign
```
Comment (tkyi) [inline lib/helper.js]:
```suggestion
    // might have more data, continue fetching
```

**Review by DekusDenial — APPROVED** [review]
LGTM, looking forwards to seeing the query option for agg duration

File: lib/helper.js (hunk 2 above)
Comment (d2lam) [reply lib/helper.js:65]: it's from `list` methods so should always be an array?

**Review by tkyi — APPROVED** [review]

**Pull request merged**